app/src/main/java/com/buzzardparking/buzzard/states/UserState.java

```diff
@@ -1,33 +1,53 @@
 package com.buzzardparking.buzzard.states;
 
 import android.content.Context;
+import android.support.annotation.Nullable;
 
 import com.buzzardparking.buzzard.activities.MapActivity;
 import com.buzzardparking.buzzard.models.AppState;
+import com.buzzardparking.buzzard.models.Spot;
 import com.buzzardparking.buzzard.util.BottomSheetManager;
 import com.buzzardparking.buzzard.util.CameraManager;
+import com.buzzardparking.buzzard.util.OnClient;
 import com.buzzardparking.buzzard.util.OnMap;
+import com.buzzardparking.buzzard.util.OnPermission;
 import com.buzzardparking.buzzard.util.PlaceManager;
+import com.google.android.gms.common.api.GoogleApiClient;
 import com.google.android.gms.maps.GoogleMap;
 
 /**
  * {@link UserState}: indicates the state a user is currently at.
  * TODO: revisit whether it's a good idea to implement the OnMap listener here
  */
-public abstract class UserState implements OnMap.Listener {
+public abstract class UserState implements OnMap.Listener, OnClient.Listener,  OnPermission.Listener {
```

Comment:
We needed to add all of the listeners because sometimes the map was ready but the client, and permissions weren't there yet.